Tests/PersonalityInsightsV3Tests/PersonalityInsightsTests.swift

```diff
@@ -0,0 +1,170 @@
+
+    /** Wait for expectations. */
+    func waitForExpectations() {
+        waitForExpectations(timeout: timeout) { error in
+            XCTAssertNil(error, "Timeout")
+        }
+    }
+    
+    // MARK: - Positive Tests
+    
+    /** Analyze the text of Kennedy's speech. */
+    func testProfile() {
+        let description = "Analyze the text of Kennedy's speech."
+        let expectation = self.expectation(description: description)
+        
+        personalityInsightsV3.getProfile(version: version,
+                                         fromText: kennedySpeech,
+                                         failure: failWithError) { profile in
+                                            for preference in profile.personality {
+                                                XCTAssertEqual("Openness", preference.name)
+                                                break
+                                             }
+                                            expectation.fulfill()
+        }
+
+        waitForExpectations()
+    }
+    
+    /** Analyze content items. */
+    func testContentItem() {
+        let description = "Analyze content items."
+        let expectation = self.expectation(description: description)
+
+        let contentItem = PersonalityInsightsV3.ContentItem(
+            id: "245160944223793152",
+            created: 1427720427,
+            updated: 1427720427,
+            contentType: "text/plain",
+            language: "en",
+            content: kennedySpeech,
+            parentID: "",
+            reply: false,
+            forward: false
+        )
+
+        let contentItems = [contentItem, contentItem]
+        personalityInsightsV3.getProfile(version: version,
+                                         fromContentItems: contentItems,
+                                         failure: failWithError) { profile in
+                                            if let behaviors = profile.behavior {                                            for behavior in behaviors {
+                                                XCTAssertNotNil(behavior.trait_id)
+                                                }
+                                                expectation.fulfill()
+                                            }
+        }
+        waitForExpectations()
+    }
+
+    // MARK: - Negative Tests
+    
+    /** Test getProfile() with text that is too short (less than 100 words). */
+    func testProfileWithShortText() {
+        let description = "Try to analyze text that is too short (less than 100 words)."
+        let expectation = self.expectation(description: description)
+
+        let failure = { (error: Error) in
+            expectation.fulfill()
+        }
+
+        personalityInsightsV3.getProfile(
+            version: version,
+            fromText: mobyDickIntro,
+            failure: failure,
+            success: failWithResult
+        )
+        waitForExpectations()
+    }
+}
```

Comment:
I think we should expand these tests to get slightly better coverage.

1. Service Function Coverage: There's three public-facing functions in the `PersonalityInsights` class--one for text input, one for HTML input, and one for content items input. It's good to make sure we have tests that exercise each public-facing function. So we should probably add a test with HTML content.

2. Default/Optional Parameters: Whenever there are default or optional parameters in public-facing functions, it's usually a good idea to test them both ways. For example, one of the public-facing functions is `getProfile(version:fromText:acceptLanguage:contentLanguage:rawScores:consumptionPreferences:failure:success:)`. But these tests never specify a value for `acceptLanguage`, `contentLanguage`, `rawScores`, or `consumptionPreferences`, so we can't be confident that these parameters will work as expected. It would be helpful to have two tests of that function: one that uses the default parameters (which we have now) and one that specifies _all_ of the parameters. (It should also set `rawScores` and `consumptionPreferences` to `true`, and we should use assert statements to verify that those additional values are set in the response.)

3. Checking Results: It's difficult to decide how to verify the results. On one extreme, we could write assert statements for every single property. On the other extreme, we could not avoid assert statements and just assume success when there are no JSON parsing errors. It's really difficult to find the right balance, but we should use at least some assert statements--enough to feel confident that the response is correct. I'll leave it up to you to decide what (if any) assert statements you think should be used to feel confident about the results. One good rule-of-thumb though, is that there should be at least one test that asserts each optional value to be non-nil (e.g. at least one test should have `XCTAssertNotNil(profile.consumptionPreferences)`, and any other asserts that you think are helpful to verify the consumption preferences).